src/test/ui/hygiene/traits-in-scope-fixme.rs

```diff
@@ -0,0 +1,29 @@
+// check-pass
+
+#![feature(decl_macro)]
+
+mod single {
+    pub trait Single {
+        fn single(&self) {}
+    }
+
+    impl Single for u8 {}
+}
+mod glob {
+    pub trait Glob {
+        fn glob(&self) {}
+    }
+
+    impl Glob for u8 {}
+}
+
+macro gen_imports() {
+    use single::Single;
+    use glob::*;
+}
+gen_imports!();
+
+fn main() {
+    0u8.single(); // FIXME, should be an error, `Single` shouldn't be in scope here
+    0u8.glob(); // FIXME, should be an error, `Glob` shouldn't be in scope here
```

Comment:
Curiously, we didn't have test cases covering imports of traits, where the traits are not produced by opaque macros (so their methods can be resolved), but the imports are produced by opaque macros (so the traits are not in scope).